backend/amd64/emit.mlp

```diff
@@ -507,7 +507,7 @@ let emit_global_label s =
   _label lbl
 
 let move (src : Reg.t) (dst : Reg.t) =
-  if src.loc <> dst.loc then
+  if src.loc <> dst.loc || Proc.register_class src <> Proc.register_class dst then
```

Comment:
Why do we need this extra condition? I thought it should now be guaranteed from `reload`.

Reply:
I don't think this extra condition is necessary indeed.